core/kazoo_apps/src/kapps_notify_publisher.erl

```diff
@@ -167,20 +167,25 @@ maybe_ignore_failure(_) -> 'false'.
 
 %% @private
 %% @doc try to find account id in different part of payload(copied from teletype_util)
--spec find_account_id(api_terms()) -> api_binary().
+-spec find_account_id(api_terms()) -> api_ne_binary().
+find_account_id(Req) when is_list(Req) ->
+    find_account_id(Req, fun props:get_first_defined/2);
 find_account_id(Req) ->
-    props:get_first_defined([<<"account_id">>
-                            ,[<<"account">>, <<"_id">>]
-                            ,<<"pvt_account_id">>
-                            ,<<"_id">>, <<"id">>
-                            ,<<"Account-ID">>
-                            ,[<<"details">>, <<"account_id">>]
-                            ,[<<"Details">>, <<"Account-ID">>]
-                            ,[<<"details">>, <<"custom_channel_vars">>, <<"account_id">>]
-                            ,[<<"Details">>, <<"Custom-Channel-Vars">>, <<"Account-ID">>]
-                            ]
-                           ,Req
-                           ).
+    find_account_id(Req, fun kz_json:get_first_defined/2).
+
+find_account_id(Req, Get) ->
+    Get([<<"account_id">>
+        ,[<<"account">>, <<"_id">>]
+        ,<<"pvt_account_id">>
+        ,<<"_id">>, <<"id">>
+        ,<<"Account-ID">>
+        ,[<<"details">>, <<"account_id">>]
+        ,[<<"Details">>, <<"Account-ID">>]
+        ,[<<"details">>, <<"custom_channel_vars">>, <<"account_id">>]
+        ,[<<"Details">>, <<"Custom-Channel-Vars">>, <<"Account-ID">>]
+        ]
```

Comment:
Out of curiosity, I tried a horse comparison:
```
kapps_notify_publisher:find_account_id in 0.585478s
kapps_notify_publisher:fai in 0.512621s
```
where find_account_id is the existing and fai implements your suggestion. Repeat factor was 1000000.  5.85478e-7 and 5.12621e-7 per invocation...so I would say this is a meaningless change anyway.